plugins/inputs/azure_monitor/azure_monitor_test.go

```diff
@@ -0,0 +1,774 @@
+          "data": [
+            {
+              "timeStamp": "2021-11-05T10:00:00Z",
+              "total": 9065573.0,
+			  "average": 8501235.0
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "id": "/subscriptions/subscription_id/resourceGroups/azure-rg/providers/Microsoft.Storage/storageAccounts/azuresa1/providers/Microsoft.Insights/metrics/Transactions",
+      "type": "Microsoft.Insights/metrics",
+      "name": {
+        "value": "Transactions",
+        "localizedValue": "Transactions"
+      },
+      "unit": "Count",
+      "timeseries": [
+        {
+          "metadatavalues": [],
+          "data": [
+            {
+              "timeStamp": "2021-11-05T10:00:00Z",
+              "total": 5.0,
+    		  "average": 4.0
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  "namespace": "Microsoft.Storage/storageAccounts",
+  "resourceregion": "eastus"
+}
+`
+
+var resourceTarget2MetricValues = `
+{
+  "cost": 0,
+  "timespan": "2021-11-05T10:00:00Z/2021-11-05T11:01:00Z",
+  "interval": "PT1M",
+  "value": [
+	{
+      "id": "/subscriptions/subscription_id/resourceGroups/azure-rg/providers/Microsoft.Storage/storageAccounts/azuresa2/providers/Microsoft.Insights/metrics/Ingress",
+      "type": "Microsoft.Insights/metrics",
+      "name": {
+        "value": "Ingress",
+        "localizedValue": "Ingress"
+      },
+      "displayDescription": "The amount of storage used by the storage account. For standard storage accounts, it's the sum of capacity used by blob, table, file, and queue. For premium storage accounts and Blob storage accounts, it is the same as BlobCapacity or FileCapacity.",
+      "unit": "Bytes",
+      "timeseries": [
+        {
+          "metadatavalues": [],
+          "data": [
+            {
+              "timeStamp": "2021-11-05T10:00:00Z",
+              "minimum": 200.0,
+			  "maximum": 200.0
+            },
+			{
+              "timeStamp": "2021-11-05T10:00:00Z",
+              "minimum": 190.0,
+			  "maximum": 210.0
+            },
+			{
+              "timeStamp": "2021-11-05T10:00:00Z",
+              "minimum": 180.0,
+			  "maximum": 220.0
+            },
+			{
+              "timeStamp": "2021-11-05T10:00:00Z",
+              "minimum": 150.0,
+			  "maximum": 225.0
+            },
+			{
+              "timeStamp": "2021-11-05T10:00:00Z",
+              "minimum": 125.0,
+			  "maximum": 250.0
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  "namespace": "Microsoft.Storage/storageAccounts",
+  "resourceregion": "eastus"
+}
+`
+
+var resourceMetricDefinitionsBody = `
+{
+  "value": [
+    {
+      "name": {
+        "value": "UsedCapacity",
+        "localizedValue": "Used capacity"
+      }
+    },
+	{
+      "name": {
+        "value": "Transactions",
+        "localizedValue": "Transactions"
+      }
+	},
+	{
+	  "name": {
+        "value": "Ingress",
+        "localizedValue": "Ingress"
+      }
+	}
+  ]
+}
+`
+
+var (
```

Comment:
This looks like you reuse this in the tests. Please don't do this. Setup the tests "standalone" to avoid side-effects.